libraries/AP_HAL/examples/TetrisGame/TetrisGame.cpp

```diff
@@ -0,0 +1,382 @@
+    SCHED_TASK(game_update,              5,   1000),
+    SCHED_TASK(proc_cmd,                 5,   1800),
+    SCHED_TASK(gameover_chck,           10,   1000),
+    SCHED_TASK(rc_update,               10,   1000),
+};
+
+void GameTetris::rc_update(void)
+{
+    bool changed = false;
+    uint8_t nchannels = hal.rcin->num_channels();
+    if (nchannels > MAX_CHANNELS) {
+        nchannels = MAX_CHANNELS;
+    }
+    for (uint8_t i=0; i<nchannels; i++) {
+        uint16_t v = hal.rcin->read(i);
+        if (last_value[i] != v) {
+            changed = true;
+            last_value[i] = v;
+        }
+        if (i > max_channels) {
+            max_channels = i;
+        }
+    }
+
+    cmd = 0x00;
+    if (changed) {
+        for (uint8_t i=0; i<max_channels; i++) {
+            if (i == JOY_LEFTRIGHT) {
+                if (last_value[i] > 1600) {
+                    cmd = 'd';
+                }
+                if (last_value[i] < 1400) {
+                    cmd = 'a';
+                }
+            }
+
+            if (i == JOY_CHGDOWN) {
+                if (last_value[i] > 1600) {
+                    cmd = ' ';
+                }
+                if (last_value[i] < 1400) {
+                    cmd = 's';
+                }
+            }
+        }
+    }
+}
+
+void GameTetris::ins_update(void)
+{
+    ins.update();
+}
+
+void GameTetris::game_update(void)
+{
+    if (!tetris.gameover) {
+        count++;
+        if (count%1 == 0) {
+            tetris.tetris_print(&tetris);
+        }
+        if (count%7 == 0) {
+            tetris.tetris_gravity(&tetris);
+            tetris.tetris_check_lines(&tetris);
+        }
+
+        if (!(hal.console->available() <= 0)) {
+            cmd = hal.console->read();
+        }
+    }
+}
+
+void GameTetris::gameover_chck(void)
+{
+    static bool terminate = false;
+    if (!terminate && tetris.gameover) {
+        terminate = true;
+        tetris.tetris_print(&tetris);
+        hal.console->printf("*** GAME OVER ***\n");
+        tetris.tetris_clean(&tetris);
+    }
+}
+
+void GameTetris::proc_cmd(void)
+{
+    switch (cmd) {
+        case 'a':
+            tetris.x--;
+            if (tetris.tetris_hittest(&tetris))
+                tetris.x++;
+            break;
+        case 'd':
+            tetris.x++;
+            if (tetris.tetris_hittest(&tetris))
+                tetris.x--;
+            break;
+        case 's':
+            tetris.tetris_gravity(&tetris);
+            break;
+        case ' ':
+            tetris.tetris_rotate(&tetris);
+            break;
+    }
+    cmd = 0x00;
+}
+
+void Tetris::tetris_init(Tetris *tetris,int _w,int _h)
+{
+    int x_aux, y_aux;
+    tetris->level = 1;
+    tetris->score = 0;
+    tetris->gameover = 0;
+    tetris->w = _w;
+    tetris->h = _h;
+    size_t sizecnt = sizeof(char*) * _w;
+    tetris->game = (char**)malloc(sizecnt);
+    sizecnt = sizeof(char) * _h;
+    for (x_aux = 0; x_aux < _w; x_aux++) {
```

Comment:
reduce the scope of x_aux to the loop by doing directly initialisation inside the for loop for (int32_t x_aux = 0;

Reply:
it is valid also for the reste for the pr ;-P, I won't write it everywhere